corehq/apps/reports/standard/sms.py

```diff
@@ -758,7 +758,10 @@ def get_queryset(self):
             data = data.filter(event_status_filter)
 
         if self.phone_number_filter:
-            data = data.filter(messagingsubevent__sms__phone_number__contains=self.phone_number_filter)
+            if self.phone_number_filter.isdigit():
+                data = data.filter(messagingsubevent__sms__phone_number__contains=self.phone_number_filter)
+            else:
+                data = data.filter(messagingsubevent__email__recipient_address__contains=self.phone_number_filter)
```

Comment:
nit: we would need to index that `recipient_address`. Can't say we have already, because I don't see the model

Reply:
https://github.com/dimagi/commcare-hq/pull/27788/files#diff-995f41f9391acbad11297718f04d841cR2870